### PR TITLE
Rework the dts generation

### PR DESCRIPTION
**Pull request by Andarist**
fixes https://linear.app/replay/issue/PRO-740

### CODE REVIEW

File: packages/jest/package.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
     "@jest/test-result": "^27.5.1",
     "@jest/types": "^27.5.1",
     "@replay-cli/pkg-build": "workspace:^",
+    "@replay-cli/shared": "workspace:^",
```
Comment (Andarist) [inline packages/jest/package.json:40]: We get a confusing error if a transitive bundled dependency isn't declared here. While we need this situation to error for regular external dependencies, I think we could compute potential bundled dependencies recursively here:
https://github.com/replayio/replay-cli/blob/229636edcd16a9a3dc11465b447763eff59c34cf/scripts/pkg-build/src/bin.ts#L29-L33

I imagine that this recursive search should include **dependencies** (unlike the top-level one above that looks at `devDependencies`)

File: packages/test-utils/src/reporter.ts [hunk 2]
```diff
@@ -264,7 +262,7 @@ class ReplayReporter<TRecordingMetadata extends UnstructuredMetadata = Unstructu
     schemaVersion: string,
     config?: ReplayReporterConfig<TRecordingMetadata>
   ) {
-    setUserAgent(`${pkgJson.name}/${pkgJson.version}`);
+    // setUserAgent(`${pkgJson.name}/${pkgJson.version}`);
```
Comment (Andarist) [inline packages/test-utils/src/reporter.ts:265]: @bvaughn https://github.com/replayio/replay-cli/pull/575 moved `test-utils` to a dev dependencies of other packages - so it's now a bundled dependency of them. We should add `setUserAgent` to all of those `test-utils` dependents, right?

Comment (bvaughn) [reply packages/test-utils/src/reporter.ts:265]: Yeah that would make more sense now that test-utils is inlined

Comment (bvaughn) [reply packages/test-utils/src/reporter.ts:265]: IMO we should just delete this line though rather than commenting it out

**Review by bvaughn — APPROVED** [review]
Stamping to unblock but didn't really read this.

**Pull request merged**